### DISCUSSION TITLE
convert post to someone else's timezone

### DISCUSSION THREAD

**Discussion by Willyfrog** [mattermost/community#49] [Fun Ideas]
at mattermost we have people from all over the world and everytime I'm talking with someone in a DM and want to say some time, I have to resort to an external website to figure out what's the time in the other person timezone.

so being able to do /timezone @anotheruser my post goes here and have this time written in it 10:00

will post (asuming 10:00 is 14:00 in @anotheruser 's timezone)

> my post goes here and have this time written in it 14:00

bonus point: no need to at-mention in DMs, and in GMs post multiple times with timezon by the side

**Comment by esethna** [comment]
Someone had a plugin for something similar to this! Where it would convert any times you entered in posts for the receiving user into their timezone. Maybe @jespino?  

We definitely need a way to inform both the sending user and receiving user that the time is converted (or show both to both users)

**Reply by M-ZubairAhmed** [reply]
I submitted this in one of the MM Hackathon 
![ds](https://user-images.githubusercontent.com/17708702/102206468-aff4dd00-3ef2-11eb-8139-3366a8d7aaa2.gif)

**Reply by M-ZubairAhmed** [reply]
https://github.com/mattermost/mattermost-hackathon-dec2020/blob/main/hackathon-submissions/m-zubairahmed-dynamic-time-conversion.md

**Reply by Willyfrog** [reply]
That's great! Have you submitted for review in community?

Does it only work on DMS or regular channels as well?

Although I'd keep the original time available in case it doesn't make sense.

**Reply by M-ZubairAhmed** [reply]
@Willyfrog yes it was for hackathon so i did just for DMs :P , 
yes i think idea of keeping original time is a really good idea, and on hover translated.

**Reply by cnu1812** [reply]
Awesome @M-ZubairAhmed

**Reply by Willyfrog** [reply]
>  yes i think idea of keeping original time is a really good idea, and on hover translated.

that would be a dream! 💯

**Reply by chetanyakan** [reply]
Can we use the tooltip override component in this plugin to allow the user seeing the message to view it in their own timezone?

**Reply by M-ZubairAhmed** [reply]
@chetanyakan yes i think its not polished, i just hacked together a prototype for the MM hackathon that time, @jfrerich helped me to execute it :)

**Reply by esethna** [reply]
@M-ZubairAhmed yes now I remember, awesome work! We should totally get this in the product built on top of the timezone feature (with a bit of UX review)

**Reply by M-ZubairAhmed** [reply]
@esethna that would be amazing. Please let me when its started so even i can contribute. 
Last time i didnt got a chance to implement Custom statuses feature which i originally proposed (may be team had already thought about it) and build in my first MM hackathon. 😄 May be i wasnt paying attention to tickets

this was my submission back then : https://github.com/mattermost/mattermost-hackathon-nov2019/blob/master/hackathon-submissions/m-zubairahmed-set-custom-status.md
![dd](https://user-images.githubusercontent.com/17708702/69648051-7db3db80-1062-11ea-8a88-6500fb018442.gif)

**Reply by esethna** [reply]
Ya thanks @M-ZubairAhmed, your work definitely inspired us to push the feature even further! We will certainly loop you in if we take your concept further regarding timezones! :)
